src/mbgl/mtl/buffer_resource.cpp

```diff
@@ -92,7 +92,12 @@ void BufferResource::update(const void* data, std::size_t updateSize, std::size_
     if (updateSize > 0) {
         if (buffer) {
             if (auto* content = static_cast<uint8_t*>(buffer->contents())) {
-                std::memcpy(content + offset, data, updateSize);
+                // Until we can be sure that the buffer is not still in use to render the
+                // previous frame, replace it with a new buffer instead of updating it.
+                auto& device = context.getBackend().getDevice();
+                if (auto newBuffer = NS::TransferPtr(device->newBuffer(content, size, usage))) {
+                    buffer = std::move(newBuffer);
```

Comment:
If I get this right we're getting a new `buffer` with the old `content` here, but the `data` patch is not being applied?